core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -251,6 +251,14 @@ public static ElasticsearchParseException newConfigurationException(String proce
         return processors;
     }
 
+    public static TemplateService.Template compileTemplate(String processorType, String processorTag, String propertyName, String propertyValue, TemplateService templateService) {
+        try {
+            return templateService.compile(propertyValue);
+        } catch (Exception e) {
+            throw ConfigurationUtils.newConfigurationException(processorType, processorTag, propertyName, e.getMessage());
+        }
```

Comment:
Ok, thought I could re-use some headers configuration stuff and enough information would make its way through the current exception's message